pipeline/src/main/java/BC_GCP_TOOLS.java

```diff
@@ -150,53 +152,85 @@ private Integer list_bucket( final String bucket, List< String > lst )
 		return res;
 	}
 
-    private Boolean download_to_file( final String bucket, final String key, final String dst_filename )
+	private void write_lock( File f )
+	{
+		BufferedWriter writer = null;
+		try
+		{
+			writer = new BufferedWriter( new FileWriter( f ) );
+			writer.write( "locked" );
+		}
+		catch( Exception e ) { e.printStackTrace(); }	
+		finally { try{ writer.close(); } catch( Exception e ) { e.printStackTrace(); } }
+	}
+
+	private void delete_lock( File f )
+	{
+		try	{ f.delete(); }	
+		catch( Exception e ) { e.printStackTrace(); }
+	}
+
+    private boolean download_to_file( final String bucket, final String key, final String dst_filename )
     {
-        Boolean res = false;
-        try
-        {
-            Storage.Objects.Get obj = storage.objects().get( bucket, key );
-            if ( obj != null )
-            {
-                obj.getMediaHttpDownloader().setDirectDownloadEnabled( true );
+        boolean res = BC_UTILS.create_paths_if_neccessary( dst_filename );
+		if ( res )
+		{
+			File f_lock = new File( String.format( "%s.lock", dst_filename ) );
```

Comment:
Looks a little racy. Is there a Java lockfile API?

Reply:
[FileLock ](https://docs.oracle.com/javase/7/docs/api/java/nio/channels/FileLock.html) comes to mind. FWIW, I was mostly focused in getting the latest code changes merged into the main trunk.

Is the code ready for a code review? @wraetz